optuna/storages/_cached_storage.py

```diff
@@ -295,12 +276,11 @@ def set_trial_values(self, trial_id: int, values: Sequence[float]) -> None:
             cached_trial = self._get_cached_trial(trial_id)
             if cached_trial is not None:
                 self._check_trial_is_updatable(cached_trial)
-                updates = self._get_updates(trial_id)
                 cached_trial.values = values
-                updates.values = values
+                self._backend.set_trial_values(trial_id, values=values)
                 return
```

Comment:
This is not a strong opinion but I personally recommend to delete these two lines to eliminate code duplications. The same applies for `set_trial_intermediate_values`, `set_trial_user_attr`, `set_trial_system_attr`.